tests/test_auth_service.py

```
@@ -0,0 +1,116 @@
+    store.write_text(json.dumps({"username": "admin"}), encoding="utf-8")
+    with pytest.raises(AuthStoreError):
+        AuthService(store, default_username="admin", default_password="pw")
+
+
+def test_unreadable_store_fails_closed(tmp_path):
+    store = _store(tmp_path)
+    store.write_text("ok-ignore", encoding="utf-8")
+    store.chmod(0o000)
+    try:
+        with pytest.raises(AuthStoreError):
+            AuthService(store, default_username="admin", default_password="pw")
```

Ruff (0.15.17)

[error] Possible hardcoded password assigned to argument: "default_password"

(S106)